src/fw-logs/fw-string-formatter.cpp

```diff
@@ -135,16 +127,57 @@ namespace librealsense
                                 s << "Protocol Error recognized!\nImproper log message received: " << source_temp
                                     << ", invalid parameter: " << exp_replace_it->second << ".\n The range of supported values is \n";
                                 for_each(vec.begin(), vec.end(), [&s](kvp& entry) { s << entry.first << ":" << entry.second << " ,"; });
-                                std::cout << s.str().c_str() << std::endl;;
+                                LOG_WARNING( s.str() );
                             }
                             source_temp = destTemp;
                         }
                     }
                 }
             }
 
-            *dest = source_temp;
+            return source_temp;
+        }
+
+        std::string fw_string_formatter::convert_param_to_string( const param_info & info, const uint8_t * param_start ) const
```

Comment:
It seems the name of this method is correct only for the case that the type is STRING.
Please find a more fitting name, or explain

Reply:
This function takes a `uint8_t*` buffer and converts the data to a string,
The data may be a string, and only a cast to `char` is needed and in might also be a binary representation of a number (of various size). This also converts the binary number (e.g. `int`) to a string representation of the number.